spec/acceptance/smb_spec.rb

```diff
@@ -0,0 +1,366 @@
+require 'acceptance_spec_helper'
+
+RSpec.describe 'SMB sessions and SMB modules' do
+  include_context 'wait_for_expect'
+
+  RHOST_REGEX = /\d+\.\d+\.\d+\.\d+:\d+/
+
+  TESTS = {
+    smb: {
+      target: {
+        session_module: "auxiliary/scanner/smb/smb_login",
+        type: 'SMB',
+        platforms: [:linux, :osx, :windows],
+        datastore: {
+          global: {},
+          module: {
+            username: ENV.fetch('SMB_USERNAME', 'acceptance_tests_user'),
+            password: ENV.fetch('SMB_PASSWORD', 'acceptance_tests_password'),
+            rhost: ENV.fetch('SMB_RHOST', '127.0.0.1'),
+            rport: ENV.fetch('SMB_RPORT', '445'),
+          }
+        }
+      },
+      module_tests: [
+        {
+          name: "post/test/smb",
+          platforms: [:linux, :osx, :windows],
+          targets: [:session],
+          skipped: false,
+        },
+        {
+          name: "auxiliary/scanner/smb/smb_lookupsid",
+          platforms: [:linux, :osx, :windows],
+          targets: [:session, :rhost],
+          skipped: false,
+          lines: {
+            all: {
+              required: [
+                "GROUP=None",
+                "USER=nobody",
+                "PIPE(LSARPC) LOCAL",
+              ],
+            },
+          }
+        },
+        {
+          name: "auxiliary/scanner/smb/pipe_auditor",
+          platforms: [:linux, :osx, :windows],
+          targets: [:session, :rhost],
+          skipped: false,
+          lines: {
+            all: {
+              required: [
+                /Pipes: (\\([a-zA-Z]*)(, )?)*/,
+              ],
+              known_failures: [
+                /Inaccessible named pipe:/,
+                /The server responded with an unexpected status code: STATUS_OBJECT_NAME_NOT_FOUND/,
+              ]
+            },
+          }
+        },
+        {
+          name: "auxiliary/scanner/smb/smb_enumshares",
+          platforms: [:linux, :osx, :windows],
+          targets: [:session, :rhost],
+          skipped: false,
+          lines: {
+            all: {
+              required: [
+                "modifiable - (DISK)",
+                "readonly - (DISK)",
+                "IPC$ - (IPC|SPECIAL) IPC Service",
+              ],
+            },
+          }
+        },
```

Comment:
If these are flakey because we need to update them to use ruby_smb, we can comment them out for now - and as we migrate modules to use ruby_smb we can bring back these tests as part of the ruby_smb support PR